submissions/Harnytskyi/friends-app/js/script.js

```diff
@@ -0,0 +1,163 @@
+const GRID = document.getElementById("allFriends");
+const CHECKBOX_MAN = document.getElementById("gender-male");
+const CHECKBOX_WOMAN = document.getElementById("gender-female");
+const SORT_ALPHABET_ASCEND = document.getElementById("a-z");
+const SORT_ALPHABET_DESCEND = document.getElementById("z-a");
+const SEARCH_INPUT = document.getElementById('search');
+const SORT_AGE_ASCEND = document.getElementById("0-9");
+const SORT_AGE_DESCEND = document.getElementById("9-0");
+const FILTER_SORT = document.getElementById("filter-sort");
+const FILTER_GENDER = document.getElementById("filter-gender");
+let friends = [];
+let gender = {
+    male: true,
+    female: true
+}
+
+let selectedFriends = [...friends];
+function filterByGender() {
```

Comment:
If you will pass `value` of selected radio button, you will have a possibility to write:

```jsx
selectedFriends = friends.filter(item => item.gender == value)
```

Instead of all these checks. And because of this, you will remove your `let gender` as redundant, because the state of selected filters kept in HTML right now ;)

Reply:
I agree with you that the use of `radio` in this case simplifies the program. 
I only need three radio buttons: "Man", "Woman" and "All". 
But I decided to use this solution because there are only two checkboxes and you can see which ones are selected at once, and it can be scaled and used in tasks with many filters.

Reply:
Please, do not resolve conversations where you have shared your opinion and did not receive an answer. :)

First thing - you will not have a case where you will need to scale this functionality - requirements will never change.

Second, your solution is very imperative and not scalable. You will need to check the state of all checkboxes with your `if ... else`  logic which will lead to very concrete, not abstract, code. Such code very hard to maintain, because on every change you will add more and more `if ... else` lines. And this will lead to another changes, which also will lead to other changes.

So, if you *really* want to keep your checkboxes, find a more abstract way to do this.
Or, considering that requirements will never change, just use `radio` buttons.